### PR TITLE
Vampire fixes!

### LINKED ISSUE #586: Misleading text colour for Vampire ability (Dominate)
The ability noted in the title, Dominate, uses the same colour for it's message to the 'victim' as the OOC colour for Moderators. This can naturally be very confusing if you've learned to filter out OOC chat with your eyes, and with how the preceding message is coloured red.

Preferable fixes would be Red + Bold to be noticeable and match the preceding message, or Purple + Italics to match the messages from the Alien whisper ability and to be very visibly unique.

### PR DESCRIPTION
**Pull request by skull132**
Fixes #586
Fixes bloodheal not handling external organs properly.
Refactors the code with new macros and shit.

Did I miss anything?

### CODE REVIEW

File: code/game/gamemodes/vampire/vampire_helpers.dm [hunk 1]
```diff
@@ -82,23 +82,23 @@
 	if (T.mind)
 		if (T.mind.assigned_role == "Chaplain")
 			if (notify)
-				src << "<span class='warning'>Your connection with the Veil is not strong enough to effect a man as devout as them.</span>"
+				to_chat(src, "<span class='warning'>Your connection with the Veil is not strong enough to effect a man as devout as them.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_helpers.dm]: Should be affect, not effect here.

File: code/game/gamemodes/vampire/vampire_helpers.dm [hunk 2]
```diff
@@ -82,23 +82,23 @@
...
-			src << "<span class='warning'>You lack the power interact with mechanical constructs.</span>"
+			to_chat(src, "<span class='warning'>You lack the power interact with mechanical constructs.</span>")
 		return 0
 
 	if (account_loyalty_implant)
 		for (var/obj/item/weapon/implant/loyalty/I in T)
 			if (I.implanted)
 				if (notify)
-					src << "<span class='warning'>You feel [T.name]'s mind unreachable due to forced loyalty.</span>"
+					to_chat(src, "<span class='warning'>You feel [T.name]'s mind unreachable due to forced loyalty.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_helpers.dm]: "mind is unreachable", maybe?

File: code/game/gamemodes/vampire/vampire.dm [hunk 3]
```diff
@@ -21,4 +21,4 @@
 	if(!help)
 		help = "Error loading help (file /ingame_manuals/vampire.html is probably missing). Please report this to server administration staff."
 
-	user << browse(help, "window=vampire_help;size=600x500")
+	show_browser(user, help, "window=vampire_help;size=600x500")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire.dm]: ~~nanoui when~~

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 4]
```diff
@@ -536,19 +540,19 @@
...
 
-		var/list/organs = get_damaged_organs()
+		var/list/organs = get_damaged_organs(1, 1)
 		if (organs.len)
 			// Heal an absurd amount, basically regenerate one organ.
 			heal_organ_damage(50, 50)
 			blood_used += 12
 
-		var/list/emotes_lookers = list("[src.name]'s skin appears to liquefy for a moment, sealing up their wounds.",
+		var/static/list/emotes_lookers = list("[src.name]'s skin appears to liquefy for a moment, sealing up their wounds.",
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: Maybe just `[src]` here so BYOND inserts "the" if needed?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 5]
```diff
@@ -597,31 +605,33 @@
 		return
 
 	if (!(vampire.status & VAMP_FULLPOWER))
-		src << "<span class='notice'>You begin peering into [T.name]'s mind, looking for a way to gain control.</span>"
+		to_chat(src, "<span class='notice'>You begin peering into [T.name]'s mind, looking for a way to gain control.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[T]`?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 6]
```diff
@@ -597,31 +605,33 @@
...
 		return
 
+	command = sanitizeSafe(command, extra = 0)
+
 	admin_attack_log(src, T, "used dominate on [key_name(T)]", "was dominated by [key_name(src)]", "used dominate and issued the command of '[command]' to")
 
-	T << "<span class='warning'>You feel a strong presence enter your mind. For a moment, you hear nothing but what it says, and are compelled to follow its direction:</span><br><span class='notice'><b>[command]</b></span>"
-	src << "<span class='notice'>You command [T.name], and they will obey.</span>"
+	show_browser(T, "<center>You feel a strong presence enter your mind. For a moment, you hear nothing but what it says, <b>and are compelled to follow its direction without question or hesitation:</b><br>[command]</center>", "window=vampiredominate")
+	to_chat(src, "<span class='notice'>You command [T.name], and they will obey.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[T]`?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 7]
```diff
@@ -635,45 +645,45 @@
...
 		return
 
 	if (vampire.status & VAMP_DRAINING)
-		src << "<span class='warning'>Your fangs are already sunk into a victim's neck!</span>"
+		to_chat(src, "<span class='warning'>Your fangs are already sunk into a victim's neck!</span>")
 		return
 
 	visible_message("<span class='danger'>[src.name] tears the flesh on their wrist, and holds it up to [T.name]. In a gruesome display, [T.name] starts lapping up the blood that's oozing from the fresh wound.</span>", "<span class='warning'>You inflict a wound upon yourself, and force them to drink your blood, thus starting the conversion process.</span>")
-	T << "<span class='warning'>You feel an irresistable desire to drink the blood pooling out of [src.name]'s wound. Against your better judgement, you give in and start doing so.</span>"
+	to_chat(T, "<span class='warning'>You feel an irresistable desire to drink the blood pooling out of [src.name]'s wound. Against your better judgement, you give in and start doing so.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[src]`?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 8]
```diff
@@ -635,45 +645,45 @@
...
 
 	if (!do_mob(src, T, 50))
 		visible_message("<span class='danger'>[src.name] yanks away their hand from [T.name]'s mouth as they're interrupted, the wound quickly sealing itself!</span>", "<span class='danger'>You are interrupted!</span>")
 		return
 
-	T << "<span class='danger'>Your mind blanks as you finish feeding from [src.name]'s wrist.</span>"
+	to_chat(T, "<span class='danger'>Your mind blanks as you finish feeding from [src.name]'s wrist.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[src]`?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 9]
```diff
@@ -692,15 +702,15 @@
...
+		to_chat(src, "<span class='warning'>No suitable targets.</span>")
 		return
 
 	var/mob/living/carbon/human/T = input(src, "Select Victim") as null|mob in victims
 
 	if (!vampire_can_affect_target(T))
 		return
 
-	src << "<span class='notice'>You infect [T.name] with a deadly disease. They will soon fade away.</span>"
+	to_chat(src, "<span class='notice'>You infect [T.name] with a deadly disease. They will soon fade away.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[T]`?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 10]
```diff
@@ -834,29 +844,29 @@
...
-				src << "<span class='notice'>[denial_response]</span>"
+				to_chat(src, "<span class='notice'>[denial_response]</span>")
 				return
 
 			vampire_thrall.remove_antagonist(T.mind, 0, 0)
 			qdel(draining_vamp)
 			draining_vamp = null
 		else
-			src << "<span class='warning'>You feel corruption running in [T.name]'s blood. Much like yourself, \he[T] is already a spawn of the Veil, and cannot be Embraced.</span>"
+			to_chat(src, "<span class='warning'>You feel corruption running in [T.name]'s blood. Much like yourself, \he[T] is already a spawn of the Veil, and cannot be Embraced.</span>")
```
Comment (Lohikar) [inline code/game/gamemodes/vampire/vampire_powers.dm]: `[T]`?

**Comment by Fire-and-Glory** [comment]
Can I use quotes on github? Idk

"Did I miss anything":

I don't see any PR's linked to this, so it may or may not still be a thing.

https://github.com/Aurorastation/Aurora.3/issues/547

**Comment by skull132** [comment]
I'm not sure I want to handle that in this PR, @Fire-and-Glory , as I'm still uncertain as to how to handle vampire cloning without redoing some of their core mechanics and adding a few more powers. Though I may implement a basic fix of, "ur still a vampire kek" or "no vampire for u, buddy".

**Comment by OneOneThreeEight** [comment]
Cultists are still cultists and wizzers are still wizzers when they get cloned. I don't think the latter keeps their spells anymore though.

**Comment by skull132** [comment]
> I don't think the latter keeps their spells anymore though.

I'm pretty confident this is a runtime.

**Pull request merged**